File: 647. Palindromic Substrings/countSubstrings.java
class Solution {
    public int isPalindrom(String s,int l,int h){

        int cnt = 0;
        while(l>=0 && h<s.length() && s.charAt(l--)==s.charAt(h++)) cnt++;
        return cnt;
    }
    public int countSubstrings(String s) {
        
        int res = 0;
        int n = s.length();
        for(int i=0;i<n;i++){

            res += isPalindrom(s,i,i);

            res += isPalindrom(s,i,i+1);
        }

        return res;
    }
}

File: 647. Palindromic Substrings/countSubstrings.cpp
class Solution
{
public:
    int solve(string s, int l, int h)
    {
        int res = 0;
        while (l >= 0 && h < s.size() && s[l--] == s[h++])
            res++;

        return res;
    }
    int countSubstrings(string s)
    {

        int ans = 0;
        int n = s.size();
        for (int i = 0; i < n; i++)
        {

            // for odd length palindrom
            ans += solve(s, i, i);
            // for even length palindrom
            ans += solve(s, i, i + 1);
        }
        return ans;
    }
};